### DISCUSSION TITLE
Is the Liquorix kernel compatible with Spiral Linux?

### DISCUSSION THREAD

**Discussion by birds-swim** [SpiralLinux/SpiralLinux-project#472] [Q&A] [answered]
![Screenshot_20240829-211518~2.png](https://github.com/user-attachments/assets/14205eee-fb43-4aa8-9f87-2ef667c68d5e)

https://liquorix.net

I've read the Arch Wiki on zRAM and I still can't figure out how zRAM and Liquorix play nicely or don't play nicely together. It seems that Liquorix uses lzo to compress swap while zRAM uses ztsd. Is that right? Am I right?

Are these two packages in direct or indirect conflict with one another?

I want to use this specific kernel for my gaming rig. I plan to live stream, edit photos and videos.

If they are in conflict, is there another custom kernel I could use?

**Comment by geckolinux** [comment]
Hi there, yep, Liquorix works with SpiralLinux, I've used it myself. Just a reminder that anything that works on a vanilla Debian installation will also work on SpiralLinux.

> It seems that Liquorix uses lzo to compress swap while zRAM uses ztsd. Is that right?

The zRAM compression algorithm is defined by the distribution in `/etc/default/zramswap` , which is Zstd by default in SpiralLinux.

**Reply by birds-swim** [reply]
Do they fight each other over who gets to swap? I wish I knew more about this stuff. Then my questions would be more on-target.

They don't conflict with each other in any way? Do I need to install lzo?

Do I really get to have my cake and eat it too??

**Reply by geckolinux** [reply, marked as answer]
Hi there, notice that zRAM isn't the same as zSwap: https://askubuntu.com/a/472227

LZO and LZ4 and Zstd are compression methods (think ZIP vs RAR) that are built into the kernel for the purpose of swap compression.

If you don't have a swap file or swap partition then zSwap won't get enabled.

According to [this](https://askubuntu.com/questions/471912/zram-vs-zswap-vs-zcache-ultimate-guide-when-to-use-which-one/472227#comment1057555_472227) comment, zRAM and zSWAP shouldn't be used concurrently.

To disable zRAM: `sudo systemctl disable zramswap.service && sudo systemctl stop zramswap.service`

To disable zSWAP if the Liquorix kernel enables it by default: https://wiki.archlinux.org/title/Zswap#Toggling_zswap

**Reply by birds-swim** [reply]
Last question. Just for clarification, if I TURN OFF zswap with `zswap.enabled=0` in my kernel parameters in my GRUB config, that will return Spiral back to its intended design which is using zram. Is that correct?

**Reply by geckolinux** [reply]
That's correct.

**Reply by birds-swim** [reply]
Awesome! Gotta check if zram-swap somehow got installed without me paying attention and remove it if necessary. Don't want that. Spiral Linux is already glorious, I don't need zram and zram-swap fighting each other.
